onnxruntime/core/optimizer/dropout_elimination.cc

```diff
@@ -25,6 +26,27 @@ bool EliminateDropout::SatisfyCondition(const Graph& graph, const Node& node, co
     return false;
   }
 
+#ifdef ENABLE_TRAINING
+  // allow Dropout elimination when:
+  //    1. ratio input is an initializer of 0
+	//    2. ratio input is not a graph input, so it cannot be overridden
+
+  // support opset 12 and above for ort training
+  if (graph_utils::MatchesOpSinceVersion(node, {12})) {
+    if (graph_utils::IsGraphInput(graph, node.InputDefs()[1])) {
+      return false;
+    }
+    const ONNX_NAMESPACE::TensorProto* initializer = graph.GetConstantInitializer(node.InputDefs()[1]->Name(), true);
+    if (!initializer) {
+      return false;
+    }
+    Initializer ratio(*initializer, graph.ModelPath());
+    if (*ratio.data<float>() > 0.f) {
```

Comment:
data can also be in float16, double ect... 
is this the standard way to read the Initializer value?

Reply:
Took example from relu_clip_fusion.cc, seems to be the case.